featuretools/tests/primitive_tests/test_transform_features.py

```diff
@@ -1570,22 +1578,77 @@ def test_cfm_with_lag_and_non_nullable_column(pd_es):
         trans_primitives=[lag_primitive],
         cutoff_time=cutoff_times,
     )
-
-    # Non nullable
     assert fm["NUMERIC_LAG(datetime, value, periods=5)"].head(periods).isnull().all()
     assert fm["NUMERIC_LAG(datetime, value, periods=5)"].isnull().sum() == periods
-    # Nullable
+
     assert "NUMERIC_LAG(datetime, value_2, periods=5)" in fm.columns
+
+    assert "NUMERIC_LAG(datetime, products.rating, periods=5)" in fm.columns
     assert (
         fm["NUMERIC_LAG(datetime, products.rating, periods=5)"]
         .head(periods)
         .isnull()
         .all()
     )
 
-    assert "NUMERIC_LAG(datetime, products.rating, periods=5)" in fm.columns
+
+def test_cfm_with_lag(pd_es):
+    # fill nans so we can use non nullable numeric logical type in the EntitySet
```

Comment:
I think we might want to consider using a different dataframe here that contains all the major logical types we are concerned with: `Double`, `Integer`, `IntegerNullable`, `Boolean`, `BooleanNullable` and `Categorical` just to make sure we have everything covered and we get the expected column types in the feature matrix.

Reply:
rewrote this test to add the above types to the dataframe and check for their existence